dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/codec/ExchangeCodec.java

```diff
@@ -175,25 +172,35 @@ protected Object decodeBody(Channel channel, InputStream is, byte[] header) thro
             Request req = new Request(id);
             req.setVersion(Version.getProtocolVersion());
             req.setTwoWay((flag & FLAG_TWOWAY) != 0);
-            if ((flag & FLAG_EVENT) != 0) {
-                req.setEvent(Request.HEARTBEAT_EVENT);
-            }
+            Object data = null;
             try {
                 ObjectInput in = CodecSupport.deserialize(channel.getUrl(), is, proto);
-                Object data;
                 if (req.isHeartbeat()) {
                     data = decodeHeartbeatData(channel, in);
                 } else if (req.isEvent()) {
                     data = decodeEventData(channel, in);
                 } else {
                     data = decodeRequestData(channel, in);
                 }
-                req.setData(data);
             } catch (Throwable t) {
                 // bad request
                 req.setBroken(true);
                 req.setData(t);
             }
+            if (!req.isBroken()) {
+                if ((flag & FLAG_EVENT) != 0) {
+                    if (data != null && data.equals(Request.READONLY_EVENT)) {

```

Comment:
`if(Request.READONLY_EVENT.equals(data)` would be more simplified i think.

Reply:
If this is bad request，`req.setData(t);` will be executed，so I think `if (!req.isBroken())` is necessary.
About `if ((flag & FLAG_EVENT) != 0)` , I think that plus him is just for insurance .
for example value of data  is "R" but it is not an event.

Reply:
Thanks for explaining this. but what I was saying is for line no 192, is 
instead of having `if (data != null && data.equals(Request.READONLY_EVENT))`  we could simplify this by just having
`if(Request.READONLY_EVENT.equals(data))`

Reply:
Thank you,It seems that you are right now, but if the value of READONLY_EVENT becomes empty in the future, this will be a hidden danger. Of course, this does not change much from the current point of view.
What are the advantages of removing the judgment?

Reply:
It is always preferable to have more simplified code if can make it 😄 . In this case if READONLY_EVENT  value become empty which is big change because it might involve changes of user code so then we can still add the judgement back (in future we might even have to add more conditions to judgement).